Common/cpp/Tools/JSScheduler.h

```diff
@@ -13,13 +13,19 @@ class JSScheduler {
  public:
   explicit JSScheduler(
       jsi::Runtime &rnRuntime,
-      const std::shared_ptr<facebook::react::CallInvoker> &jsCallInvoker)
-      : rnRuntime_(rnRuntime), jsCallInvoker_(jsCallInvoker) {}
+      const std::shared_ptr<facebook::react::CallInvoker> &jsCallInvoker,
+      std::function<void(std::function<void(jsi::Runtime &runtime)> &&callback)>
+          runtimeExecutor)
+      : rnRuntime_(rnRuntime),
+        jsCallInvoker_(jsCallInvoker),
+        runtimeExecutor_(runtimeExecutor) {}
```

Comment:
I'm not a fan of the `runtimeExecutor_` potentially being a `nullptr`. Instead of passing null as the runtimeExecutor, let's create two constructors: one for bridgeless mode and the second one for bridgefull mode. The second constructor will assign an empty lambda instead of null.

Reply:
We could even go one step further and add an enum field `mode_` with two options: `JSCallInvokerMode` and `RuntimeExecutorMode`.

Reply:
Done in [2efade7](https://github.com/software-mansion/react-native-reanimated/pull/5734/commits/2efade7e68c277b2ab8ca61c31814a910dce2703)